js/renovation/check-box.tsx

```diff
@@ -0,0 +1,199 @@
+      onValueChanged={viewModel.props.onValueChanged}
+      onClick={viewModel.onWidgetClick}
+      onInactive={viewModel.onInactive}
+      onKeyDown={viewModel.onWidgetKeyDown}
+      rtlEnabled={viewModel.props.rtlEnabled}
+      tabIndex={viewModel.props.tabIndex}
+      visible={viewModel.props.visible}
+      width={viewModel.props.width}
+      {...viewModel.restAttributes} // eslint-disable-line react/jsx-props-no-spreading
+    >
+      {name
+        ? <input ref={viewModel.submitInputRef} type="hidden" value={`${viewModel.props.value}`} name={name} />
+        : <input ref={viewModel.submitInputRef} type="hidden" value={`${viewModel.props.value}`} />}
+      <div className={CHECKBOX_CONTAINER_CLASS} ref={viewModel.contentRef}>
+        <span className={CHECKBOX_ICON_CLASS} />
+        {text && (<span className={CHECKBOX_TEXT_CLASS}>{text}</span>)}
+      </div>
+      {viewModel.props.useInkRipple
+                && <InkRipple config={inkRippleConfig} ref={viewModel.inkRippleRef} />}
+    </Widget>
+  );
+};
+
+@ComponentBindings()
+export class CheckBoxProps extends BaseWidgetProps {
+  @OneWay() activeStateEnabled?: boolean = true;
+
+  @OneWay() hoverStateEnabled?: boolean = true;
+
+  @OneWay() elementAttr?: object = {};
+
+  @OneWay() text?: string = '';
+
+  @OneWay() name?: string = '';
+
+  @TwoWay() value?: boolean | null = false;
+
+  @OneWay() useInkRipple?: boolean = false;
+
+  @Event({
+    actionConfig: { excludeValidators: ['disabled', 'readOnly'] },
+  }) onValueChanged?: (e: object) => void = (() => {});
+}
+
+export const defaultOptionRules = createDefaultOptionRules<CheckBoxProps>([{
+  // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+  device: (): boolean => devices.real().deviceType === 'desktop' && !(devices as any).isSimulator(),
+  options: { focusStateEnabled: true },
+}]);
+
+@Component({
+  defaultOptionRules,
+  jQuery: { register: true },
+  view: viewFunction,
+})
+export default class CheckBox extends JSXComponent(CheckBoxProps) {
+  @Ref() contentRef!: HTMLDivElement;
+
+  @Ref() inkRippleRef!: InkRipple;
+
+  @Ref() submitInputRef!: HTMLInputElement;
+
+  @Ref() widgetRef!: Widget;
+
+  @Method()
+  focus(): void {
+    this.widgetRef.focus();
+  }
+
+  @Effect()
+  contentReadyEffect(): void {
+    const { onContentReady } = this.props;
+    // eslint-disable-next-line  @typescript-eslint/no-non-null-assertion
+    onContentReady!({ element: this.submitInputRef.parentNode });
+  }
+
+  onActive(event: Event): void {
+    const { useInkRipple } = this.props;
+
+    useInkRipple && this.inkRippleRef.showWave({ element: this.widgetRef, event });
+  }
+
+  onInactive(event: Event): void {
+    const { useInkRipple } = this.props;
+
+    useInkRipple && this.inkRippleRef.hideWave({ element: this.widgetRef, event });
+  }
+
+  onWidgetClick(event: Event): void {
+    const { value, onValueChanged } = this.props;
+
+    this.props.value = !value;
```

Comment:
A strange string. According ideology we should not change props inside

Reply:
I agree, but how should we change the value in this case?